goalc/make/MakeSystem.cpp

```diff
@@ -329,4 +362,4 @@ bool MakeSystem::make(const std::string& target, bool force, bool verbose) {
   fmt::print("\nSuccessfully built all {} targets in {:.3f}s\n", deps.size(),
              make_timer.getSeconds());
   return true;
-}
\ No newline at end of file
+}
```

Comment:
I think as written, changing the `:arg` or adding a `:input` won't trigger a rebuild.  Maybe you should check the timestamp on the project file itself?